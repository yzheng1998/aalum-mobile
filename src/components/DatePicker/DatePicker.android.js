import React, { Component } from 'react'
import { DatePickerAndroid, View } from 'react-native'

export default class DatePicker extends Component {
  constructor(props) {
    super(props)
    this.state = {
      error: ''
    }
    const { setDate, doneOnPress, date } = props
    this.openDatePicker = async () => {
      try {
        const { action, year, month, day } = await DatePickerAndroid.open({
          date,
          mode: 'spinner'
        })
        if (action !== DatePickerAndroid.dismissedAction) {
          setDate(new Date(year, month, day))
          doneOnPress()
        } else if (action === DatePickerAndroid.dismissedAction) {
          doneOnPress()
        }
      } catch ({ code, message }) {
        this.setState({ error: `Cannot open date picker: ${message}` })
      }
    }
  }

  render() {
    return <View />
  }
}
