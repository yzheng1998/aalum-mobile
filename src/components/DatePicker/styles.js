import styled from 'styled-components'
import { themeGet } from 'styled-system'

export const DoneButton = styled.TouchableOpacity`
  height: 45px;
  width: 100%;
  justify-content: center;
  align-items: center;
  background-color: ${themeGet('colors.white')};
`

export const DoneButtonText = styled.Text`
  font-size: 16px;
  color: ${themeGet('colors.charcoal')};
  font-family: ${themeGet('fonts.bodyText')};
`

export const Container = styled.View`
  background-color: ${themeGet('colors.white')};
  position: absolute;
  align-self: center;
  bottom: 0;
  width: 100%;
`
