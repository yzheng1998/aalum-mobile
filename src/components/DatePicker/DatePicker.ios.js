import React, { Component } from 'react'
import { DatePickerIOS, View } from 'react-native'
import { DoneButton, DoneButtonText, Container } from './styles'

export default class DatePicker extends Component {
  openDatePicker = () => null

  render() {
    const { doneOnPress, date, setDate, mode, visible } = this.props
    return (
      <Container>
        {visible && (
          <View>
            <DoneButton onPress={doneOnPress}>
              <DoneButtonText>Done</DoneButtonText>
            </DoneButton>
            <DatePickerIOS mode={mode} date={date} onDateChange={setDate} />
          </View>
        )}
      </Container>
    )
  }
}
