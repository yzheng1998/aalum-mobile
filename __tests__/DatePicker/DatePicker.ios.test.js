import React from 'react'
import DatePicker from '../../src/components/DatePicker/DatePicker.ios'
import { shallow } from 'enzyme'

describe('DatePicker iOS', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<DatePicker visible={false} />)
    expect(wrapper).toMatchSnapshot()

    const mockDate = new Date('01/01/1996')
    wrapper.setProps({ visible: true, mode: 'date', date: mockDate })
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ mode: 'datetime' })
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ mode: 'time' })
    expect(wrapper).toMatchSnapshot()
  })

  test('calls doneOnPress correctly', () => {
    const onPressMock = jest.fn()
    const mockDate = new Date('01/01/1996')
    const wrapper = shallow(
      <DatePicker
        date={mockDate}
        mode="date"
        doneOnPress={onPressMock}
        visible
      />
    )
    const button = wrapper.find('doneButton')
    button.simulate('press')

    expect(onPressMock).toHaveBeenCalledTimes(1)
  })
})
