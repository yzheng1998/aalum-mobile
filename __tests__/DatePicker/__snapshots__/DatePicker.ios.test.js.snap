// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker iOS renders correctly 1`] = `<Styled(View) />`;

exports[`DatePicker iOS renders correctly 2`] = `
<Styled(View)>
  <View>
    <doneButton>
      <Styled(Text)>
        Done
      </Styled(Text)>
    </doneButton>
    <DatePickerIOS
      date={1996-01-01T05:00:00.000Z}
      mode="date"
    />
  </View>
</Styled(View)>
`;

exports[`DatePicker iOS renders correctly 3`] = `
<Styled(View)>
  <View>
    <doneButton>
      <Styled(Text)>
        Done
      </Styled(Text)>
    </doneButton>
    <DatePickerIOS
      date={1996-01-01T05:00:00.000Z}
      mode="datetime"
    />
  </View>
</Styled(View)>
`;

exports[`DatePicker iOS renders correctly 4`] = `
<Styled(View)>
  <View>
    <doneButton>
      <Styled(Text)>
        Done
      </Styled(Text)>
    </doneButton>
    <DatePickerIOS
      date={1996-01-01T05:00:00.000Z}
      mode="time"
    />
  </View>
</Styled(View)>
`;
